lib/linter.js

```diff
@@ -340,7 +340,19 @@ function modifyConfigsFromComments(filename, ast, config, ruleMapper) {
                             Object.keys(parseResult.config).forEach(name => {
                                 const ruleValue = parseResult.config[name];
 
-                                validator.validateRuleOptions(ruleMapper(name), name, ruleValue, `${filename} line ${comment.loc.start.line}`);
+                                try {
+                                    validator.validateRuleOptions(ruleMapper(name), name, ruleValue);
+                                } catch (err) {
```

Comment:
Should we decorate the thrown error object with some property to indicate it is specifically the validation error we want to report? That way, if we check for that property here, an unexpected exception (from something not directly due to schema validation) would propagate as a crash.

Reply:
Seems reasonable, although I think we already use thrown errors to communicate schema issues in other places.

Maybe it would be better to switch to a result type instead, where `validator.validateRuleOptions` would return something like `{ success: true }` or `{ success: false, error: err }`. Then a thrown error would always indicate that something went wrong.